### PR TITLE
fix(StarRating): Fix UX bug #1650

### PR DESCRIPTION
**Pull request by bobylito**
**Summary**

Fixes the ability to click even though there is no behavior triggered, when clicking on the default value and it is already selected. #1650 For reference:

**Result**

Now the default value is not clickable if it is selected:

And it also fixes a bug introduced by the highlighting PR :(

### CODE REVIEW

File: packages/react-instantsearch/src/components/StarRating.js [hunk 1]
```diff
@@ -49,6 +49,32 @@ class StarRating extends Component {
         )}
         />);
     }
+    if (isLowest && selected) {
```
Comment (vvo) [inline packages/react-instantsearch/src/components/StarRating.js]: Did you try to merge both this code path with the normal one? I am wondering if this could be a bit less verbose and only changes a into div + avoid setting onClick and href props.

Something like

```js
buildItem('a', commonProps, {href: ...});
buildItem('div', commonProps);
```

Comment (vvo) [inline packages/react-instantsearch/src/components/StarRating.js]: As we discussed via DM, here's the proposed git diff:

```
diff --git a/packages/react-instantsearch/src/components/StarRating.js b/packages/react-instantsearch/src/components/StarRating.js
index abcec5e..486e6b3 100644
--- a/packages/react-instantsearch/src/components/StarRating.js
+++ b/packages/react-instantsearch/src/components/StarRating.js
@@ -31,9 +31,10 @@ class StarRating extends Component {
     }
   }
 
-  buildItem({max, lowerBound, count, translate, createURL, isLowest}) {
+  buildItem({max, lowerBound, count, translate, createURL}) {
     const selected = lowerBound === this.props.currentRefinement.min &&
       max === this.props.currentRefinement.max;
+    const isLowestAndSelected = lowerBound === this.props.min && selected;
     const disabled = !count;
 
     const icons = [];
@@ -49,58 +50,36 @@ class StarRating extends Component {
         )}
         />);
     }
-    if (isLowest && selected) {
-      return (
-        <div {...cx(
-          'ratingLink',
-          'ratingLinkSelected',
-          disabled && 'ratingLinkDisabled')}
-           disabled={disabled}
-           key={lowerBound}
-        >
-          {icons}
-          <span {...cx(
-            'ratingLabel',
-            'ratingLabelSelected',
-            disabled && 'ratingLabelDisabled')}>
-            {translate('ratingLabel')}
-            </span>
-          <span> </span>
-          <span {...cx(
-            'ratingCount',
-            'ratingCountSelected',
-            disabled && 'ratingCountDisabled')}>
-            {count}
-          </span>
-        </div>
-      );
-    }
-    return (
-      <a {...cx(
-        'ratingLink',
-        selected && 'ratingLinkSelected',
-        disabled && 'ratingLinkDisabled')}
-         disabled={disabled}
-         key={lowerBound}
-         onClick={this.onClick.bind(this, lowerBound, max)}
-         href={createURL({lowerBound, max})}
-      >
-        {icons}
-        <span {...cx(
-          'ratingLabel',
-          selected && 'ratingLabelSelected',
-          disabled && 'ratingLabelDisabled')}>
-          {translate('ratingLabel')}
-          </span>
-        <span> </span>
-        <span {...cx(
-          'ratingCount',
-          selected && 'ratingCountSelected',
-          disabled && 'ratingCountDisabled')}>
-          {count}
+
+    const Wrapper = isLowestAndSelected ? 'div' : 'a';
+    const clickProps = isLowestAndSelected ? {} : {
+      onClick: this.onClick.bind(this, lowerBound, max),
+      href: createURL({lowerBound, max}),
+    };
+
+    return <Wrapper {...cx(
+      'ratingLink',
+      selected && 'ratingLinkSelected',
+      disabled && 'ratingLinkDisabled')}
+      disabled={disabled}
+      key={lowerBound}
+      {...clickProps}
+    >
+      {icons}
+      <span {...cx(
+        'ratingLabel',
+        selected && 'ratingLabelSelected',
+        disabled && 'ratingLabelDisabled')}>
+        {translate('ratingLabel')}
         </span>
-      </a>
-    );
+      <span> </span>
+      <span {...cx(
+        'ratingCount',
+        selected && 'ratingCountSelected',
+        disabled && 'ratingCountDisabled')}>
+        {count}
+      </span>
+    </Wrapper>;
   }
```

Comment (vvo) [inline packages/react-instantsearch/src/components/StarRating.js]: If you believe (like I do) this is a ok iteration then I can push the patch

File: packages/react-instantsearch/src/connectors/connectHighlight.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ export default createConnector({
 
   propTypes: {},
 
-  getProps() {
+  getProvidedProps() {
```
Comment (vvo) [inline packages/react-instantsearch/src/connectors/connectHighlight.js]: I am moving this in another commit inside v2 if you don't mind since it's not directly linked to the current bug. So that we can release a new beta version and updated the website.

**Comment by bobylito** [comment]
Refactoring done!

File: packages/react-instantsearch/src/components/StarRating.js [hunk 3]
```diff
@@ -49,15 +49,24 @@ class StarRating extends Component {
         )}
         />);
     }
+
+    // The last item of the list (the default item), should not
+    // be clickable if it is selected.
+    const isLastAndSelect = isLowest && selected;
+    const StarsWrapper = isLastAndSelect ? 'div' : 'a';
+    const onClickHandler = isLowest && selected ? {} : {
```
Comment (vvo) [inline packages/react-instantsearch/src/components/StarRating.js]: => isLastAndSelect?
